libraries/AP_BoardConfig/board_drivers.cpp

```diff
@@ -262,39 +260,19 @@ void AP_BoardConfig::validate_board_type(void)
 #endif
 }
 
-
-void AP_BoardConfig::check_cubeblack(void)
-{
-#if defined(HAL_CHIBIOS_ARCH_CUBEBLACK)
-    if (state.board_type != PX4_BOARD_PIXHAWK2) {
-        state.board_type.set(PX4_BOARD_PIXHAWK2);
-    }
-
-    bool success = true;
-    if (!spi_check_register("mpu9250", MPUREG_WHOAMI, MPU_WHOAMI_MPU9250)) { success = false; }
-    if (!spi_check_register("mpu9250_ext", MPUREG_WHOAMI, MPU_WHOAMI_MPU9250) &&
-        !spi_check_register("icm20602_ext", MPUREG_WHOAMI, MPU_WHOAMI_ICM20602)) { success = false; }
-    if (!(spi_check_register("lsm9ds0_ext_g", LSMREG_WHOAMI, LSM_WHOAMI_L3GD20) && 
-          spi_check_register("lsm9ds0_ext_am", LSMREG_WHOAMI, LSM_WHOAMI_LSM303D)) &&
-        !spi_check_register("icm20948_ext", INV2REG_WHOAMI, INV2_WHOAMI_ICM20948)) { success = false; }
-    if (!check_ms5611("ms5611")) { success = false; }
-    if (!check_ms5611("ms5611_ext")) { success = false; }
-
-    if (!success) {
-        config_error("Failed to init CubeBlack - sensor mismatch");
-    }
-#endif
-}
-
-
 /*
   auto-detect board type
  */
 void AP_BoardConfig::board_autodetect(void)
 {
-#if defined(HAL_CHIBIOS_ARCH_CUBEBLACK)
-    check_cubeblack();
-    return;
+#if defined(VALIDATE_BOARD)
```

Comment:
I'd prefer the macro to be HAL_VALIDATE_BOARD, just to keep the namespaces a bit separate

Reply:
> I like the concept, thanks Sid!
> I've suggested a small macro name change, plus can you please run:
> ./Tools/scripts/configure_all.py --no-bl --build
> and check that all boards still build ok? Then check bootloader builds.
> After structural changes to hwdef processing I like to check all builds

ok, I will check the builds, I thought the CI tests cover that, but I will double check.